gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionContext.java

```diff
@@ -323,22 +323,34 @@ private void buildSourceToOrderByExpressionMaps() {
     return ImmutableMap.<String, List<TimeAggregation>>builder().putAll(result).build();
   }
 
-  private void getSourceToFilterExpressionMap(Filter filter,
-                                              Map<String, List<Expression>> sourceToExpressionMap) {
+  private void getSourceToFilterExpressionMap(
+      Filter filter, Map<String, List<Expression>> sourceToFilterExpressionMap) {
     if (Filter.getDefaultInstance().equals(filter)) {
       return;
     }
 
     if (filter.hasLhs()) {
       // Assuming RHS never has columnar expressions.
       Expression lhs = filter.getLhs();
-      Map<String, List<Expression>> map = getDataSourceToExpressionMap(List.of(lhs));
-      sourceToExpressionMap.putAll(map);
```

Comment:
`putAll` is bad! It does not merge the map, it replaces the existing keys in the map